app/services/refresh_materialized_views.rb

```diff
@@ -39,37 +42,44 @@ def self.tz
 
   delegate :tz, :set_last_updated_at, to: self
 
-  def refresh
-    # LatestBloodPressuresPerPatientPerMonth should be refreshed before
-    # LatestBloodPressuresPerPatientPerQuarter and LatestBloodPressuresPerPatient
+  V1_MATVIEWS = %w[
+    LatestBloodPressuresPerPatientPerMonth
+    LatestBloodPressuresPerPatient
+    LatestBloodPressuresPerPatientPerQuarter
+    BloodPressuresPerFacilityPerDay
+    PatientRegistrationsPerDayPerFacility
+    MaterializedPatientSummary
+  ].freeze
+  V2_MATVIEWS = %w[
+    ReportingPipeline::PatientBloodPressuresPerMonth
+    ReportingPipeline::PatientStatesPerMonth
+    ReportingPipeline::PatientVisitsPerMonth
+  ].freeze
+
+  # LatestBloodPressuresPerPatientPerMonth should be refreshed before
+  # LatestBloodPressuresPerPatientPerQuarter and LatestBloodPressuresPerPatient
+  def refresh_v1
     ActiveRecord::Base.transaction do
       ActiveRecord::Base.connection.execute("SET LOCAL TIME ZONE '#{tz}'")
-
-      benchmark_and_statsd("LatestBloodPressuresPerPatientPerMonth") do
-        LatestBloodPressuresPerPatientPerMonth.refresh
-      end
-
-      benchmark_and_statsd("LatestBloodPressuresPerPatient") do
-        LatestBloodPressuresPerPatient.refresh
-      end
-
-      benchmark_and_statsd("LatestBloodPressuresPerPatientPerQuarter") do
-        LatestBloodPressuresPerPatientPerQuarter.refresh
-      end
-
-      benchmark_and_statsd("BloodPressuresPerFacilityPerDay") do
-        BloodPressuresPerFacilityPerDay.refresh
-      end
-
-      benchmark_and_statsd("PatientRegistrationsPerDayPerFacility") do
-        PatientRegistrationsPerDayPerFacility.refresh
+      V1_MATVIEWS.each do |name|
+        benchmark_and_statsd(name) do
+          klass = name.constantize
+          klass.refresh
+        end
       end
+      set_last_updated_at
+    end
+  end
 
-      benchmark_and_statsd("MaterializedPatientSummary") do
-        MaterializedPatientSummary.refresh
+  def refresh_v2
+    ActiveRecord::Base.transaction do
+      ActiveRecord::Base.connection.execute("SET LOCAL TIME ZONE '#{tz}'")
```

Comment:
`ReportingPipeline::Matview.refresh` already takes care of the transaction and the timezone setting. We can just wrap with metrics here.

Reply:
The outer transaction is to rollback if any part of this fails I think, which is why we are doing it w/ the other views as well.

The `tz` piece is also just for safety and consistency w/ the other refresh...the more I think about this the more I think we should just move all the refresh logic here and not have it in the model objects at all.  It doesn't make sense to refresh a matview in isolation, as they are all pretty tightly coupled w/ each other.

Reply:
I actually think it's better to express the dependency tree closer to the model where there's most context about it. Each view can express its own dependencies, that way the refresh logic is in one place. When we change it for tests, we change it for the job as well.

Reply:
I mean, you can't really see the dependency tree unless you look at the SQL itself.  Abstractions on abstractions here.  😁 

> I actually think it's better to express the dependency tree closer to the model where there's most context about it. Each view can express its own dependencies, that way the refresh logic is in one place.

I can't think of a way to do that that would be more clear than a method that calls the `refresh` methods in the order we need, the dependency ordering is pretty clear there.  Suggestions welcome though.

Reply:
I just meant having the same list in the patient-state model itself. I'll raise a tiny PR for this perhaps.